designs/PackageVulnerability/DotnetListPackageVulnerable.md

```diff
@@ -248,21 +239,19 @@ The following sources were used:
 
 Project `ClassLibrary1` uses the following vulnerable packages
    [netcoreapp2.0]:
-   Top-level Package              Requested  Resolved    Vulnerability
-   > My.Vulnerable.Package        2.0.0      2.0.0       https://URL-to-vuln-details
-   > My.Fixed.Package             1.1.0      1.1.0       https://URL-to-vuln-details
+   Top-level Package              Requested  Resolved    Severity   Vulnerability
+   > My.Vulnerable.Package        2.0.0      2.0.0       Low        https://URL-to-vuln-details
+   > My.Fixed.Package             1.1.0      1.1.0       Critical   https://URL-to-vuln-details
 
  > To see transitive packages, add the `--include-transitive` option.
- > To see prerelease packages, add the `--include-prerelease` option.
 ```
 
-*Combining the `--vulnerable` option with the `--offline` options is not supported. When doing so, the following error message will be presented:*
+Notes:
+* *Important! Installed packages that are deprecated or outdated but not vulnerable will not be part of this command's output.*
 
-```shell
-> dotnet list package --vulnerable --offline
+* None of the markers `(D)`, `(O)` or `(V)` are used in this report.
```

Comment:
If we use markers in the other commands (deprecated / outdated), what harm is there to add (D) & (O) to the `--vulnerable` command? This is especially helpful for those who have outdated packages that may contain a security vulnerability. Authors who may know about a vulnerability may then deprecate their package so users are unlikely to use it.

i.e.

**Outdated**
```
dotnet list package --outdated
Lists which installed packages have newer versions available.

> dotnet list package --outdated

The following sources were used:
   nuget.org - https://api.nuget.org/v3/index.json
   Local - C:\NuGet\NuGetLocal

3 packages need your attention - 2 outdated, 1 deprecated, 1 vulnerable.

Package                Requested   Resolved    Latest
EntityFramework        6.1.2       6.1.2       6.2.0
NUnit                  2.4.0       2.6.4 (D)   3.8.1  
My.Sample.Pkg          2.1.3       4.1.0 (V)   4.1.0

 > To see transitive packages, add the `--include-transitive` option.
(D): Deprecated package(s). Use 'dotnet list package --deprecated' for more info.
(V): Vulnerable package(s). Use 'dotnet list package --vulnerable' for more info.
```

**Deprecated**
```
dotnet list package --deprecated
Lists which installed packages have been deprecated.

> dotnet list package --deprecated

The following sources were used:
   nuget.org - https://api.nuget.org/v3/index.json
   Local - C:\NuGet\NuGetLocal

Project `ClassLibrary1` uses the following deprecated packages
   [netcoreapp2.0]:
   Top-level Package              Requested  Resolved    Reason                 Alternative
   > My.Legacy.Package            2.0.0      2.0.0       Legacy                 My.Awesome.Package >= 3.0.0
   > My.Buggy.Package             1.1.0      1.1.0 (O)   Critical Bugs          My.NotBuggy.Package >= 2.0.0
   > My.Deprecated.Package        3.2.1      3.2.1       Other                  My.NotBuggy.Package >= 2.0.0
   > My.CompletelyBroken.Package  0.9.0      0.9.0 (V)   Legacy, Critical Bugs  My.Awesome.Package >= 1.0.0

 > To see transitive packages, add the `--include-transitive` option.
(O): Outdated package(s). Use 'dotnet list package --outdated' for more info.
(V): Vulnerable package(s). Use 'dotnet list package --vulnerable' for more info.
```

Reply:
I'm happy to put all markers everywhere (except the default `dotnet list package` experience which is offline)--this would be consistent across report types, and IMO consistency removes confusion. My implementation of the earlier spec included this approach, which received pushback here: https://github.com/NuGet/NuGet.Client/pull/3615#issuecomment-682330207. But let's discuss here and arrive at a decision on whether D/O/V markers should be:
- nowhere
- in some places as tracked by this spec
- in some places with adjustments to this spec
- everywhere (except the default `dotnet list package` experience)

Reply:
`(O)` as an outdated marker is super weird tbh. 
I think using `O` for anything other than `output` is misleading, legends be damned. 

That being said, I don't have a different suggestion.

Reply:
Do we know CLI UX experts? There may be a common pattern we're unaware of that we can leverage.

Reply:
> Do we know CLI UX experts?

Not if anyone has that role but I think Kathleen might be as close as we can get to that :)

Reply:
@loic-sharma Not necessarily a common pattern, but I think it's one that may want to be considered in the future.

i.e. https://www.npmjs.com/package/npm-check

Just looking at the experiences we're trying to match here, markers really don't make sense in this context. The goal is that if a user typed in any command `--vulnerable`, `--deprecated`, or `--outdated`, that they would ultimately see a footer of "How to fix it" via an `audit / audit fix` command.

Right now, we're trying to just keep the user typing these commands until they understand what is all vulnerable, deprecated, and outdated. The goal really is to get them to run our `audit` command, and ultimately fix everything in one command with an `audit fix` command.

I'm in favor of just removing markers given the contention on them entirely & removing the footnotes as well. Thoughts?

Reply:
That all makes sense to me--let's go with that plan. If we want markers and footers later, we can add them after agreeing on what we want. For now, let's remove the blockers. Thanks. 👍

Reply:
Here's a thought of what that looks like:

## Vulnerable

dotnet list package --vulnerable
Lists which installed packages have known vulnerabilities.

```
> dotnet list package --vulnerable

The following sources were used:
   nuget.org - https://api.nuget.org/v3/index.json
   Local - C:\NuGet\NuGetLocal

Project `ClassLibrary1` uses the following vulnerable packages
   [netcoreapp2.0]:
   Top-level Package              Requested  Resolved    Severity   Vulnerability
   > My.Vulnerable.Package        2.0.0      2.0.0       Low        https://URL-to-vuln-details
   > My.Fixed.Package             1.1.0      1.1.0       Critical   https://URL-to-vuln-details

 > To see transitive packages, add the `--include-transitive` option.
 > Run `dotnet nuget audit fix` to fix them, or `dotnet nuget audit` for more details
```

## Deprecated

dotnet list package --deprecated
Lists which installed packages have been deprecated.

```
> dotnet list package --deprecated

The following sources were used:
   nuget.org - https://api.nuget.org/v3/index.json
   Local - C:\NuGet\NuGetLocal

Project `ClassLibrary1` uses the following deprecated packages
   [netcoreapp2.0]:
   Top-level Package              Requested  Resolved    Reason                 Alternative
   > My.Legacy.Package            2.0.0      2.0.0       Legacy                 My.Awesome.Package >= 3.0.0
   > My.Buggy.Package             1.1.0      1.1.0       Critical Bugs          My.NotBuggy.Package >= 2.0.0
   > My.Deprecated.Package        3.2.1      3.2.1       Other                  My.NotBuggy.Package >= 2.0.0
   > My.CompletelyBroken.Package  0.9.0      0.9.0       Legacy, Critical Bugs  My.Awesome.Package >= 1.0.0

 > To see transitive packages, add the `--include-transitive` option.
 > Run `dotnet nuget audit fix` to fix them, or `dotnet nuget audit` for more details
```

## Outdated

dotnet list package --outdated
Lists which installed packages have newer versions available.

```
> dotnet list package --outdated

The following sources were used:
   nuget.org - https://api.nuget.org/v3/index.json
   Local - C:\NuGet\NuGetLocal

Project `ClassLibrary1` uses the following deprecated packages
   [netcoreapp2.0]:
   Top-level Package        Requested   Resolved    Latest
   > EntityFramework        6.1.2       6.1.2       6.2.0
   > NUnit                  2.4.0       2.6.4       3.8.1  
   > My.Sample.Pkg          2.1.3       4.1.0       4.1.0

 > To see transitive packages, add the `--include-transitive` option.
 > Run `dotnet nuget audit fix` to fix them, or `dotnet nuget audit` for more details
```

Reply:
@JonDouglas --thanks. Note thought that the transitive footers are still in these samples, so they can all be considered gone as well, right?

Reply:
@drewgillies Ah sorry, yes it's late. Let me remove those.

Reply:
I don't think there's any warrant in having them without markers for the moment.

Reply:
>I don't think there's any warrant in having them without markers

Could you please clarify this comment--we want the reports without markers (except `(A)`--we can leave those in), right?

Reply:
To clarify, we don't need the `--include-transitive` footer as it'll be in the `--help` section. It doesn't really add any value being in each footer.